### PR TITLE
workflows: add autopublish

### PR DESCRIPTION
**Pull request by p-linnane**
Bringing this in line with the main `homebrew-cask` repo.

### CODE REVIEW

File: .github/workflows/autopublish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run automerge
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
+        run: |
+          brew pr-automerge --verbose \
+                            --publish \
+                            --tap homebrew/cask \
```
Comment (Rylan12) [inline .github/workflows/autopublish.yml:31]:
```suggestion
                            --tap homebrew/cask-drivers \
```

**Comment by Rylan12** [comment]
Also, can these be synced using https://github.com/Homebrew/homebrew-cask/blob/master/.github/workflows/sync-templates-and-ci-config.yml

**Comment by bevanjkay** [comment]
@Rylan12 and we pass the tap name
dynamically?

**Comment by Rylan12** [comment]
I think with `{ github.repository }` you can. It's been a while since I poked GitHub actions though so maybe I'm misremembering the syntax.

For the `if` conditions to only run on taps I think we can use `if: startsWith(github.repository, 'Homebrew/')`

File: .github/workflows/autopublish.yml (hunk 1 above)
Comment (p-linnane) [reply .github/workflows/autopublish.yml:31]: Oversight on my part. Thanks!

**Comment by p-linnane** [comment]
Closing in favor of https://github.com/Homebrew/homebrew-cask/pull/138802.

**Pull request closed without merging**